ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/QuestionAnsweringModel.java

```diff
@@ -47,50 +63,230 @@ public void warmUp(Predictor predictor, String modelId, MLModelConfig modelConfi
+
+        List<Output> initialOutputs = getPredictor().batchPredict(List.of(initialInput));
+        for (Output output : initialOutputs) {
+            ModelTensors tensors = parseModelTensorOutput(output, null);
+            allHighlights.addAll(extractHighlights(tensors));
+        }
+    }
+
+    private void processOverflowChunks(
+        String question,
+        String context,
+        SentenceHighlightingQATranslator translator,
+        List<Map<String, Object>> allHighlights
+    ) throws TranslateException {
+        Encoding encodings = translator.getTokenizer().encode(question, context);
+        Encoding[] overflowEncodings = encodings.getOverflowing();
+
+        if (overflowEncodings == null || overflowEncodings.length == 0) {
+            return;
+        }
+
+        List<Input> overflowInputs = createOverflowInputs(question, context, overflowEncodings.length);
+        processOverflowInputs(overflowInputs, allHighlights);
+    }
+
+    private List<Input> createOverflowInputs(String question, String context, int numOverflowChunks) {
+        List<Input> overflowInputs = new ArrayList<>();
+        for (int i = 0; i < numOverflowChunks; i++) {
+            Input chunkInput = new Input();
+            chunkInput.add(MLInput.QUESTION_FIELD, question);
+            chunkInput.add(MLInput.CONTEXT_FIELD, context);
+            chunkInput.add(HIGHLIGHTING_MODEL_CHUNK_NUMBER_KEY, String.valueOf(i + 1));
+
+            overflowInputs.add(chunkInput);
+        }
+        return overflowInputs;
+    }
+
+    private void processOverflowInputs(List<Input> overflowInputs, List<Map<String, Object>> allHighlights) throws TranslateException {
+        try {
+            processOverflowInputsBatch(overflowInputs, allHighlights);
+        } catch (IllegalArgumentException e) {
+            log.info("Batch processing of chunks failed. Processing chunks individually: {}", e.getMessage());
+            processOverflowInputsIndividually(overflowInputs, allHighlights);
+        }
+    }
+
+    private void processOverflowInputsBatch(List<Input> overflowInputs, List<Map<String, Object>> allHighlights) throws TranslateException {
+        List<Output> overflowOutputs = getPredictor().batchPredict(overflowInputs);
+        for (Output output : overflowOutputs) {
+            try {
+                ModelTensors tensors = parseModelTensorOutput(output, null);
+                allHighlights.addAll(extractHighlights(tensors));
+            } catch (Exception e) {
+                log.warn("Error processing output from chunk", e);
+            }
+        }
+    }
+
+    private void processOverflowInputsIndividually(List<Input> overflowInputs, List<Map<String, Object>> allHighlights)
+        throws TranslateException {
+        for (int i = 0; i < overflowInputs.size(); i++) {
+            processOverflowChunkWithBatch(i + 1, overflowInputs.get(i), allHighlights);
```

Comment:
can we add more tests and more comments to explain the code logic and reason?

Reply:
not familiar with TranslateException, have you test the error code and make sure it's a proper 400x error?